packages/core/src/app/shipping/Shipping.test.tsx

```diff
@@ -500,6 +501,56 @@ describe('Shipping step', () => {
             expect(checkoutService.updateBillingAddress).toHaveBeenCalled();
             expect(screen.getByText(payments[0].config.displayName)).toBeInTheDocument();
         });
+
+        it('goes back to the shipping step as a guest and updates the shipping address form correctly', async () => {
+            checkout.use(CheckoutPreset.CheckoutWithShippingAndBilling);
+
+            checkout.updateCheckout(
+                'put',
+                '/checkouts/xxxxxxxxxx-xxxx-xxax-xxxx-xxxxxx/consignments/consignment-1',
+                {
+                    ...checkoutWithShippingAndBilling,
+                },
+            );
+
+            render(<CheckoutTest {...defaultProps} />);
+
+            await checkout.waitForPaymentStep();
+
+            await userEvent.click(screen.getAllByRole('button', {name: 'Edit'})[1]);
+
+            const randomAddress1 = JSON.parse(JSON.stringify({
+                firstName: faker.name.firstName(),
+                lastName: faker.name.lastName(),
+                address1: faker.address.streetAddress(),
+                city: faker.address.city(),
+                countryCode: 'CC',
+                stateOrProvince: 'dummy state',
+                postalCode: faker.address.zipCode(),
+            }));
+
+            await checkout.fillAddressForm(randomAddress1);
+
+            expect((checkoutService.updateShippingAddress as any).mock.calls.slice(-1)[0][0]).toEqual(
+                expect.objectContaining(randomAddress1)
+            );
+
+            const randomAddress2 = JSON.parse(JSON.stringify({
+                firstName: faker.name.firstName(),
+                lastName: faker.name.lastName(),
+                address1: faker.address.streetAddress(),
+                city: faker.address.city(),
+                countryCode: 'AU',
+                stateOrProvinceCode: faker.helpers.arrayElement(['NSW', 'VIC', 'QLD', 'TAS']),
+                postalCode: faker.address.zipCode(),
+            }));
+
+            await checkout.fillAddressForm(randomAddress2);
+
+            expect((checkoutService.updateShippingAddress as any).mock.calls.slice(-1)[0][0]).toEqual(
+                expect.objectContaining(randomAddress2)
+            );
```

Comment:
We have not previously tested the state/province dropdown menu.
This test will ensure that both the dropdown menu and the text box state/province scenarios are covered.